src/IO/S3Common.cpp

```diff
@@ -214,8 +322,9 @@ namespace S3
         client_configuration.updateSchemeAndRegion();
 
         Aws::Auth::AWSCredentials credentials(access_key_id, secret_access_key);
+
         return std::make_shared<Aws::S3::S3Client>(
-            std::make_shared<S3AuthSigner>(client_configuration, std::move(credentials), std::move(headers)),
+            std::make_shared<S3AuthSigner>(client_configuration, std::move(credentials), std::move(headers), use_environment_credentials),
             std::move(client_configuration), // Client configuration.
```

Comment:
BTW, the order of evaluation of function arguments is 'unspecified' (i.e. `f(++i, ++i)`). So this `client_configuration` arg must be taken care of.

Reply:
+